### PR TITLE
Initial draft of excluder blog post

### PR DESCRIPTION
**Pull request by JeffreyRStevens**
* [x] I have read the Content Guidelines.
* [x] I have read the Technical Guidelines.
* [x] I used or followed the R Markdown or Markdown template.
* [x] I have followed the Style Guide.
* [x] I created or updated my author metadata with correct folder name.
* [x] I have added relevant tags after browsing existing tags (including "community" tag).
* [x] I have added the "tech notes" tag if this is a technote.
* [x] I ran `roblog::ro_lint_md()` on index.md (optional).
* [x] I ran `roblog::ro_check_urls()` on index.md (optional).
* [x] I ran a spell-check on index.md.
* [x] I have added the tags - Software Peer Review, my-packagename.
* [x] I have added the package-version YAML tag.
* [x] I have added acknowledgement of the reviewers' work (with links to reviewers).
* [x] I have added a link to the software peer review thread.

### CODE REVIEW

**Comment by JeffreyRStevens** [comment]
In my local view, the output in R code chunks are not showing up in a new line after the code ends. They show up in the same line as the last line of code.

File: content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+* **IP address country** (`ip`)<sup>*</sup>: 
+If the data are non-anonymous, IP addresses are available, which can determine the country of origin for the respondent. 
+excluder can detect IP addresses outside of a specified country. 
+Note, however, that VPNs can use IP addresses outside of the country of origin, so you must weigh the possibility of false alarms in detecting IP addresses in other countries.
+* **Geolocation in US** (`location`)<sup>*</sup>: 
+Qualtrics determines geolocation (latitude and longitude) through GPS or IP addresses. 
+excluder can detect if location originates from outside of the United States but cannot detect this for other countries at the moment.
+* **Duplicate IP address** (`duplicates`)<sup>*</sup>: 
+Automated bots may spam your survey from the same IP address. 
+excluder detect if the same IP address shows up repeatedly in your data.
```
Comment (steffilazerte) [inline content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd:92]:
```suggestion
excluder detects if the same IP address shows up repeatedly in your data. 
```

File: content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+Each mark function will output this kind of message letting us know how many rows meet each metadata criterion.
+Second, we now have a new column called `exclusion_preview` appended to the data frame.
+Notice that the first two rows of that column have `preview` values while the rest of the rows are blank.
+We have now marked which rows meet this metadata criterion.
+
+Of course, you may want to mark your data for multiple metadata criteria. 
+To do this, you simply have to connect multiple mark functions with a pipe (either [base R's `|>`](https://rdrr.io/r/base/pipeOp.html) or [magrittr's %>%](https://magrittr.tidyverse.org/reference/pipe.html)). 
+So if we want to mark preview data and all rows where the completion duration is less than 200 seconds or more than 600 seconds, we just chain these together. 
+In fact, we can also pull the data frame out of functions as well. 
+Note that I'm going to show just a subset of columns and rows for clarity (ignore from `dplyr::select()` on).
```
Comment (steffilazerte) [inline content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd:138]: This is something I struggle with when teaching a lot. I think it's better to explain the function than to suggest ignoring it. Try:
```suggestion
Note that for clarity, I use the [`select()`](https://dplyr.tidyverse.org/reference/select.html) function from [dplyr](https://dplyr.tidyverse.org/reference/select.html) to show just a subset of columns.
```

File: content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+As we exclude, the messages tell us how many rows are returned from each functions, so we can see how the data frame gets smaller at each step. 
+While the order of the functions shouldn't influence the final data frame output, it might make sense to use criteria with many exclusions early to speed processing of subsequent exclusions.
+
+## Deidentifying data
+
+Congratulations---we have now removed all of our problematic data!
+However, those data frames may still have identifying information included in them (e.g., IP addresses and geolocation).
+Before we pass on the data set to others or post it publicly, we probably want to remove those identifying columns. 
+excluder has a `deidentify()` function that quickly removes either all metadata columns.
```
Comment (steffilazerte) [inline content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd:225]: In Part 1, you mentioned that qualtRics also has options for deidentifying the data. Is it worth adding a sentence or two here how the two methods are different?

Comment (steffilazerte) [inline content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd:225]:
```suggestion
excluder has a `deidentify()` function that quickly removes all metadata columns.
```

File: content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+```{r}
+deidentify(qualtrics_text, strict = FALSE) %>%
+  head()
+```
+
+## Conclusion
+
+Before the fun part of analyzing data, we often need to perform some more tedious steps. The [qualtRics](https://docs.ropensci.org/qualtRics/) package is a huge help in automating the import of data into R. 
+As we learned in Part 1 of this post, after connecting to the Qualtrics API, we can quickly (and powerfully) import our Qualtrics data with the `fetch_survey()` function.
```
Comment (steffilazerte) [inline content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd:242]:
```suggestion
As we learned in [Part 1](blog/2022/08/02/working-with-qualtrics-data-importing/), after connecting to the Qualtrics API, we can quickly (and powerfully) import our Qualtrics data with the [`fetch_survey()`](https://docs.ropensci.org/qualtRics/reference/fetch_survey.html) function.
```

File: content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+{{< figure src = "happy_data.jpg" width = "600" alt = "A woman sitting on a black couch with a laptop is looking left and smiling. Photo by Brooke Cagle on Unsplash." class = "center" >}}
+```
+
+If you have ideas for improvements or find bugs, please [submit an issue](https://github.com/ropensci/excluder/issues) on the [excluder GitHub repo](https://github.com/ropensci/excluder/).
+
+## Acknowledgments
+
+In developing the excluder package, I thank [Francine Goh](https://orcid.org/0000-0002-7364-4398) and Billy
+Lim for comments on an early version of the package, as well as the
+insightful feedback during [peer review](https://github.com/ropensci/software-review/issues/455) from [rOpenSci](/) editor [Mauro
```
Comment (steffilazerte) [inline content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd:257]:
```suggestion
insightful feedback during [peer review](https://github.com/ropensci/software-review/issues/455) from rOpenSci editor [Mauro
```

File: content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+## Conclusion
+
+Before the fun part of analyzing data, we often need to perform some more tedious steps. The [qualtRics](https://docs.ropensci.org/qualtRics/) package is a huge help in automating the import of data into R. 
+As we learned in Part 1 of this post, after connecting to the Qualtrics API, we can quickly (and powerfully) import our Qualtrics data with the `fetch_survey()` function.
+Once those data are imported, we can use the [excluder](https://docs.ropensci.org/excluder/) package to deal with potentially problematic online survey data based on respondent metadata using the mark, check, and exclude functions. 
+Then we can deidentify our data set before continuing to work with it or passing it on to others.
+Combined, these two packages can improve your data workflow by automating import and early cleaning processes, leaving the fun part of data processing up to you!
+
+```{=html}
+{{< figure src = "happy_data.jpg" width = "600" alt = "A woman sitting on a black couch with a laptop is looking left and smiling. Photo by Brooke Cagle on Unsplash." class = "center" >}}
```
Comment (steffilazerte) [inline content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd:248]: I think the attribution should be visible, try adding it with the `caption = ` tag.

File: content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+  - community
+  - data munging
+  - excluder
+  - packages
+  - Software Peer Review
+  - survey data
+package_version: 0.4.0
+description: "This post summarizes how to use the excluder package to exclude survey data."
+tweet: "Exclude problematic survey data with the excluder R package by @JeffStevensADML!"
+output: hugodown::md_document
```
Comment (steffilazerte) [inline content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd:18]: Consider adding the hex sticker for excluder as the twitterImg. Just update the `name-of-image.png` and alt text and add that file to the post folder. 
```suggestion
twitterImg: blog/2022/08/09/working-with-qualtrics-data-excluding/name-of-image.png
twitterAlt: "Alternative description of the image"
output: hugodown::md_document
```

**Review by steffilazerte — COMMENTED** [review]
Hi @JeffreyRStevens,

Again, a beautifully written post, thank you! I've made some suggestions for linking to Part 1, adding to the metadata, and a couple of wording suggestions. 

As before, my suggestions and comments are only that, so feel free to use them, modify them or ignore them :grin:

One thing to remember is that you'll want to omit the first Part from this pull request so that it doesn't conflict with the published version later on. 

The problem with the code output is related to some changes we've been making to address your original issue with figures but also a problem with another post. The current solution is to knit your Rmd file with `blogdown::build_site(build_rmd = "content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd")`, but this also requires changes made to the roweb3 main branch. 

@maelle, should Jeffrey simply merge in the main branch changes (which will be squashed later), or does he have to rebase? (@JeffreyRStevens, either way, you might as well wait until after Part 1 is published).

If you can get these revisions done by Monday August 8th, we'll be ready to publish Tuesday August 9th.

* [X] post follows Content Guidelines
* [X] post follows Style Guide
* [X] title is in Title Case
* [X] publication date is ok
* [X] alternative text of images is informative
* [ ] Twitter metadata looks ok (paste post preview link in [Twitter card validator](https://cards-dev.twitter.com/validator); might have to click twice on "Preview card")
    * This is optional, but I gave a suggestion for an image, and want to remember to test it (although the actual card validator hasn't been working lately)
* [ ] author metadata is provided with correct folder name
* [X] html not included in pull request of Rmd post
* [X] I ran `roblog::ro_lint_md()` on [[index.md](http://index.md/)](http://index.md/)
* [X] I ran `roblog::ro_check_urls()` on [[index.md](http://index.md/)](http://index.md/)
* [X] I ran a spell-check on [[index.md](http://index.md/)](http://index.md/)
* [X] YAML subject tags are ok ("tech notes" for tech notes; "community" for non-staff non-editor)
* [X] YAML package-version included
* [X] YAML subject tags - software peer review, packagename
* [X] acknowledges and links to reviewers
* [X] links to peer review thread

**Review by steffilazerte — COMMENTED** [review]
Looks good! I've made a couple of final comments, feel free to take or omit as you like.

The only thing remaining, is to figure out how to make sure that when we merge this branch in, we don't revert anything in last week's post. 

There are some conflicts that popped up as a result of merging the branches (see the conflicts notice at the bottom of the 'conversation'). Right now I think we have two options:

1. Fix the conflicts and make sure last weeks post isn't been modified
2. Copy and paste your blog post folder into a new pull request based on the up-to-date main branch. 

I feel like the second option may be cheating, but is the safest. Sorry for the confusion! I've never dealt with a multipart post so we've run into some challenges I'm unfamiliar with (and the webmaster is on vacation!)

File: content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+
+* **Qualtrics preview status** (`preview`): 
+Qualtrics records data when you [preview your survey](https://www.qualtrics.com/support/survey-platform/survey-module/preview-survey/) before deploying it. 
+You probably always want to remove preview data.
+* **Survey progress** (`progress`): 
+Qualtrics tracks how much of the survey the respondent completes. 
+excluder can detect surveys which are incomplete.
+* **Survey completion time** (`duration`): 
+Qualtrics records the number of seconds that the respondent works on the survey. 
+excluder can detect completion times that are too fast or that take too long.
```
[the remark this answers is not in the document]
Comment (JeffreyRStevens) [reply content/blog/2022-08-09-working-with-qualtrics-data-excluding/index.Rmd:85]: I prefer 'that' to 'which' here.

**Comment by steffilazerte** [comment]
Awesome! Draft tweet:

```
[blog] Working with Qualtrics Data - Part 2: Excluding Data

Dealing with messy survey data getting you down? 😟

Check out @JeffStevensADML's step-by-step guide to the excluder package for cleaning up survey data 🎉

https://ropensci.org/blog/2022/08/09/working-with-qualtrics-data-excluding/

#RStats #qualtrics #excluder

(1/2)

---

🙏 Thanks to Francine W Goh, Billy Lim, @mauro_lepore, Joeseph O'Brien, @juliasilge for feedback and reviews!

👀 Don't forget last week's post "Working with Qualtrics Data - Part 1: Importing Data" for how to *download* survey data with qualtRics!

https://ropensci.org/blog/2022/08/02/working-with-qualtrics-data-importing/

(2/2)
```

**Pull request merged**